### PR TITLE
Fix overloads, destructors, deprecation, don't output entire config file to screen

### PR DESCRIPTION
**Pull request by wxmerkt**

### CODE REVIEW

**Comment by mauricefallon** [comment]
have you tested this?

**Comment by wxmerkt** [comment]
Yes I am running it with the Spot log on Noetic

**Comment by mauricefallon** [comment]
Ok, awesome. That's pretty dedicated of you!

The improvement with undistortion on the clouds fed with Spot's odometry is clear cut.

**Comment by mcamurri** [comment]
Thank you @wxmerkt . I was aware of the annoying PCL warning but wasn't sure how to fix it properly.

File: lidar_undistortion_ros/include/lidar_undistortion_ros/ouster_undistorter_ros.hpp [hunk 1]
```diff
@@ -14,6 +14,7 @@ namespace lidar_undistortion {
 class OusterUndistorterROS : public OusterUndistorter {
 public:
  OusterUndistorterROS(ros::NodeHandle& nh);
+ virtual ~OusterUndistorterROS() = default;
```
Comment (mcamurri) [inline lidar_undistortion_ros/include/lidar_undistortion_ros/ouster_undistorter_ros.hpp]: I believe you can set it as override, the super destructor should be virtual:
`~OusterUndistorterRos() override = default;`

**Review by mcamurri — CHANGES_REQUESTED** [review]
All good

**Comment by wxmerkt** [comment]
It does make a large difference - I am using it to have "better" point clouds going into the GaitMesh reconstruction. And it improves the surface planes/normals we get.

**Comment by mauricefallon** [comment]
I dont know if you are accumulating the point clouds yourself or using elevation mapping
EM is pretty good:
https://www.dropbox.com/s/k1nxx5xfjp1k2pa/spot_elevation_mapping3.mp4?dl=0
but you need to turn off visibility clear out in the elevation mapping setting.

**Comment by mcamurri** [comment]
[jenkins test this]

**Review by mcamurri — APPROVED** [review]

**Comment by wxmerkt** [comment]
We've been accumulating point clouds ourselves but I am moving to using EM (since otherwise the floor is too sparse), thanks for the pointer :-)

The CI failure is due to a CI configuration issue - it is not extending any upstream workspace (cc @benoit-robotics):

```
--------------------------------------------------------------------------------
Profile:                     default
**Extending:                   None**
Workspace:                   /var/lib/jenkins/workspace/drs_pr_ori_drs_lidar_undistortion/catkin_ws
--------------------------------------------------------------------------------
``` 

Extending says "None" when it should read `/opt/ros/DISTRO`

**Comment by benoit-robotics** [comment]
the CI fails because the build server hasn't been entirely migrated  to Ubuntu 20, I hope to finish that soon

**Pull request merged**